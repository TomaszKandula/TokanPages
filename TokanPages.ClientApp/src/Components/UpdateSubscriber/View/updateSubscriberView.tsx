import * as React from "react";
import { AccountCircle } from "@material-ui/icons";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";
import Skeleton from "@material-ui/lab/Skeleton";
import { UpdateSubscriberStyle } from "./updateSubscriberStyle";

interface IBinding
{
    bind: IProperties;
}

interface IProperties
{
    isLoading: boolean;
    caption: string;
    formHandler: any;
    email: string;
    buttonHandler: any;
    buttonState: boolean;
    progress: boolean;
    buttonText: string;
    labelEmail: string;
}

export const UpdateSubscriberView = (props: IBinding): JSX.Element =>
{
    const classes = UpdateSubscriberStyle();

    const ActiveButton = (): JSX.Element => 
    {
        return(
            <Button fullWidth onClick={props.bind?.buttonHandler} variant="contained" 
                className={classes.button} disabled={props.bind?.progress || !props.bind?.buttonState}>
                {props.bind?.progress &&  <CircularProgress size={20} />}
                {!props.bind?.progress && props.bind?.buttonText}
            </Button>
        );
    }

    return (
        <section className={classes.section}>
            <Container maxWidth="sm">
                <Box pt={18} pb={10}>
                    <Card elevation={4}>
                        <CardContent className={classes.card}>
                            <Box mb={3} textAlign="center">
                                <AccountCircle className={classes.account} />
                                <Typography className={classes.caption}>
                                    {props.bind?.isLoading ? <Skeleton variant="text" /> : props.bind?.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        <TextField 
                                            required fullWidth onChange={props.bind?.formHandler} value={props.bind?.email} label={props.bind?.labelEmail}
                                            variant="outlined" name="email" id="email" autoComplete="email" 
                                        />
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    {props.bind?.isLoading ? <Skeleton variant="rect" /> : <ActiveButton />}
                                </Box>
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>
    );
}
