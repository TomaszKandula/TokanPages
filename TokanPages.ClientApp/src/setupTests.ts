// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
