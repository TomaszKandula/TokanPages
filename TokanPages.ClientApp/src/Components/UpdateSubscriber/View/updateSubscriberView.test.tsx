import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { UpdateSubscriberView } from "./updateSubscriberView";

describe("Test component: updateSubscriberView.", () => 
{
    it("Renders correctly '<UpdateSubscriberView />' when content is loaded.", () => 
    {
        const tree = shallow(<UpdateSubscriberView bind=
        {{
            isLoading: false,
            caption: "Update subscription email",
            formHandler: jest.fn(),
            email: "ester.exposito@gmail.com",
            buttonHandler: jest.fn(),
            buttonState: false,
            progress: false,
            buttonText: "Update",
            labelEmail: "Email address"
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
