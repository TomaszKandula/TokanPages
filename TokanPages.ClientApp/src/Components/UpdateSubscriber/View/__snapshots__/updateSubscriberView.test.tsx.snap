// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: updateSubscriberView should render correctly '<UpdateSubscriberView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      pb={10}
      pt={18}
    >
      <WithStyles(ForwardRef(Card))
        elevation={4}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card-4"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <Memo(ForwardRef(AccountCircleIcon))
              className="makeStyles-account-2"
            />
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-caption-3"
            >
              Update subscription email
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="email"
                  fullWidth={true}
                  id="email"
                  label="Email address"
                  name="email"
                  onChange={[MockFunction]}
                  required={true}
                  value="ester.exposito@gmail.com"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <Styled(MuiBox)
              my={2}
            >
              <ActiveButton
                buttonHandler={[MockFunction]}
                buttonState={false}
                buttonText="Update"
                caption="Update subscription email"
                email="ester.exposito@gmail.com"
                formHandler={[MockFunction]}
                isLoading={false}
                labelEmail="Email address"
                progress={false}
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
